controllers/remoteexecnodes_test.go

```diff
@@ -0,0 +1,377 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ytv1 "github.com/ytsaurus/yt-k8s-operator/api/v1"
+	"github.com/ytsaurus/yt-k8s-operator/pkg/consts"
+)
+
+const (
+	eventuallyWaitTime = 10 * time.Second
+	eventuallyTickTime = 500 * time.Millisecond
+)
+
+const (
+	remoteYtsaurusHostname   = "test-hostname"
+	remoteYtsaurusName       = "test-remote-ytsaurus"
+	remoteExecNodesName      = "test-remote-exec-nodes"
+	statefulSetName          = "end-test-remote-exec-nodes"
+	execNodeConfigMapName    = "yt-exec-node-config"
+	execNodeConfigMapYsonKey = "ytserver-exec-node.yson"
+)
+
+var (
+	testYtsaurusImage = "test-ytsaurus-image"
+)
+
+// TestRemoteExecNodesFromScratch ensures that remote exec nodes resources are
+// created with correct connection to the specified remote Ytsaurus.
+func TestRemoteExecNodesFromScratch(t *testing.T) {
+	h := newTestHelper(t, "remote-exec-nodes-test-from-scratch")
+	h.start()
+	defer h.stop()
+
+	remoteYtsaurusSpec := buildRemoteYtsaurus(h, remoteYtsaurusName, remoteYtsaurusHostname)
+	deployObject(h, &remoteYtsaurusSpec)
+	waitRemoteYtsaurusDeployed(h, remoteYtsaurusName)
+	nodes := buildRemoteExecNodes(h, remoteYtsaurusName, remoteExecNodesName)
+	deployObject(h, &nodes)
+	waitRemoteExecNodesDeployed(h, remoteExecNodesName)
+
+	fetchEventually(h, statefulSetName, &v1.StatefulSet{})
+
+	ysonNodeConfig := fetchConfigMapData(h, execNodeConfigMapName, execNodeConfigMapYsonKey)
+	require.NotEmpty(t, ysonNodeConfig)
+	require.Contains(t, ysonNodeConfig, remoteYtsaurusHostname)
+}
+
+// TestRemoteExecNodesYtsaurusChanges ensures that if remote Ytsaurus CRD changes its hostnames
+// remote nodes changes its configs accordingly.
+func TestRemoteExecNodesYtsaurusChanges(t *testing.T) {
+	h := newTestHelper(t, "remote-exec-nodes-test-host-change")
+	h.start()
+	defer h.stop()
+
+	remoteYtsaurus := buildRemoteYtsaurus(h, remoteYtsaurusName, remoteYtsaurusHostname)
+	deployObject(h, &remoteYtsaurus)
+	waitRemoteYtsaurusDeployed(h, remoteYtsaurusName)
+	nodes := buildRemoteExecNodes(h, remoteYtsaurusName, remoteExecNodesName)
+	deployObject(h, &nodes)
+	waitRemoteExecNodesDeployed(h, remoteExecNodesName)
+
+	ysonNodeConfig := fetchConfigMapData(h, execNodeConfigMapName, execNodeConfigMapYsonKey)
+	require.NotEmpty(t, ysonNodeConfig)
+	require.Contains(t, ysonNodeConfig, remoteYtsaurusHostname)
+
+	hostnameChanged := remoteYtsaurusHostname + "-changed"
+	remoteYtsaurus.Spec.MasterConnectionSpec.HostAddresses = []string{hostnameChanged}
+	updateObject(h, &ytv1.RemoteYtsaurus{}, &remoteYtsaurus)
+	waitRemoteYtsaurusDeployed(h, remoteYtsaurusName)
+
+	fetchAndCheckEventually(
+		h,
+		execNodeConfigMapName,
+		&corev1.ConfigMap{},
+		func(obj client.Object) bool {
+			data := obj.(*corev1.ConfigMap).Data
+			ysonNodeConfig = data[execNodeConfigMapYsonKey]
+			return strings.Contains(ysonNodeConfig, hostnameChanged)
+		},
+	)
+}
+
+// TestRemoteExecNodesImageUpdate ensures that if remote exec nodes images changes controller
```

Comment:
`If <condition>, [then] <statement>`
In English compound sentences almost always use  a comma. ".. images changes, controller..."

Reply:
Here and in other similar comments.